### ISSUE TITLE
[BUG] <title>Qwen7B最新模型和代码微调对数据格式有什么要求吗？

### ISSUE THREAD

**Issue by Coder-nlper** [QwenLM/Qwen#699]
### 是否已有关于该错误的issue或讨论？ | Is there an existing issue / discussion for this?

- [X] 我已经搜索过已有的issues和讨论 | I have searched the existing issues / discussions

### 该问题是否在FAQ中有解答？ | Is there an existing answer for this in FAQ?

- [X] 我已经搜索过FAQ | I have searched FAQ

### 当前行为 | Current Behavior

使用的多轮对话数据格式如下：
第一轮
user：xxxx
assistant：xxxx
第二轮
user：xxxx
assistant：xxxx
。。。。

使用<endoftext>作为结束符

使用以前旧版的Qwen，loss训练最终在0.1左右，得训练5epoch左右，使用学习率策略是constant，学习率5e-5，模型可以正常使用

今天换新的模型和代码。训练0.4 epoch，loss，都到0.0001了，然后停止训练，在测试集上全错。怀疑过拟合了。。。
我使用的是基座模型，按道理对数据格式要求不高吧

### 运行环境 | Environment

```Markdown
- OS:Ubuntu 20.04
- Python:3.10
- Transformers:4.34.0
- PyTorch:2.1
- CUDA (`python -c 'import torch; print(torch.version.cuda)'`):12.1
```

**Comment by micronetboy** [comment]
遇到同样的问题，我是用的 Qwen-14B-chat-int4 QLora, Loss 下降的特别快。但微调数据没有起任何作用。测试结果，和没有微调一样。

**Comment by jklj077** [comment]
根据我了解到的信息Qwen-7B与Qwen-14B-Chat-Int4的参数并未有更新。finetune.py代码有一处变更，将deivce_map从None改为了auto，这部分变更，我用finetune_qlora_single_gpu.sh (transformers 4.34) 复现微调结果似乎是无影响的，但可能对显存有限的设备有影响。可否试下将device_map设置为None能否解决该问题，这个patch https://github.com/QwenLM/Qwen/compare/main...jklj077-patch-1

**Comment by micronetboy** [comment]
试过了 设置为 None 也不行。Loss下降特别快。微调数据没有起任何作用。好像是 QLora 没起任何作用。

**Comment by micronetboy** [comment]
我用上个版本的 finetune.py 和 modeling_qwen.py，也还是不行

**Comment by micronetboy** [comment]
微调过程中：最明显的变化是 Loss 下降特别快，微调数据不起任何作用

**Comment by vcvcvnvcvcvn** [comment]
我也遇到了同样的问题，lora的时候loss飞速下降
<img width="717" alt="image" src="https://github.com/QwenLM/Qwen/assets/52404040/c416ed60-9631-42cb-909d-70d2d8e5015d">
微调数据格式应该没啥问题，就像这样：
<img width="979" alt="image" src="https://github.com/QwenLM/Qwen/assets/52404040/249fc185-4c0c-4f0a-9b85-99e4ad1a6c39">
结果调出来发现没有效果

**Comment by godcrying** [comment]
我微调也是出现loss下降很快，才训练了20 多个样本，loss直接0.004，剩下1万多的样本都没法训练了。
